### ISSUE TITLE
Feature request for webclient: Option to open EvMenu in popup

### ISSUE THREAD

**Issue by FlutterSprite** [evennia/evennia#1524]
#### Description of requested feature:
The web client allows the user to open up help files in a pop-up window - this is very good! It helps keep lengthy files from crowding out the log and reduces scrolling, and also lessens the likelihood that a player will miss anything during a helpfile binge.

It would be really great if there could be an option to open menus in a pop-up window too - long EvMenus tend to be rather spammy affairs (especially with the abundance of line breaks in the default formatting), so it would be helpful for the same reason putting helpfiles in pop-ups is.

#### Reasons for adding feature:

1. Due to the way menus are already formatted in the webclient, it may be more intuitive - by default, all the options in an EvMenu are presented as clickable links, and today's web users will probably find the most natural way of navigating them to be clicking through them. However, since all the menu text and options are posted to the logs, the clickable options remain after they're no longer relevant, leading to the unintuitive situation where clicking options from a previous menu node will throw up errors, even though they're still on screen. Presenting menus in a pop-up window that only presents the relevant node and its options would eliminate this problem.

2. Having a pop-up window with a menu in it would be a clear indicator to the player that they're currently in a menu, and as such their access to commands may be changed or limited, when it might otherwise be unclear.

3. A terminal text feed is best suited to keyboard-based command line control, while pop-up windows are better suited to pointing and clicking - since EvMenu in the web client seems to favor a pointing and clicking style of interaction, a pop-up would be a more natural way to interact with it. Of course, since it'd be an option, those who prefer the current behavior could turn it off.

#### Extra information, such as Evennia revision/repo/branch, operating system and ideas for how to solve / implement:

I imagine it'd be helpful to have an option in the EvMenu function itself to force it into the main window feed even if the 'Menus in popups' option in the client is on - since EvMenu is so flexible, there may be instances in which the developer would want to be sure it appears in the main feed.

**Comment by Griatch** [comment]
This is probably a good idea, the webclient is suitable for these kinds of experiments. I wonder if it would be an addie to wait for @chainsol's experimentations with adjustable client windows before starting to add  things like this to the client API though.

**Comment by chainsol** [comment]
Actually, sure, yes, we could make this configurable in the settings for the help. Currently the only option is to slap it into that little draggable window, but in my changed version, you could choose to output to a new pane. I could add an option to make a new window as well!

**Comment by Griatch** [comment]
@chainsol The pane would be the first option but the ability to put stuff in a closable, draggable window is always a nice feature too if not too hard to add. :)

**Comment by FlutterSprite** [comment]
Yeah! There's plenty of other default commands I'd like to see have the option to open in a pop-up as well - I think 'inventory' is definitely a good candidate for this, as well.

**Comment by chainsol** [comment]
Would an inline pop-up be better (what we have now) or an actual pop-up window you can drag/resize separately?

**Comment by j-f1** [comment]
For reference, you can see the changes I made to add a new popup type in my Ainneve PR: https://github.com/evennia/ainneve/pull/97.

**Comment by christophermstaats** [comment]
Actual popup window would be best.

**Comment by FlutterSprite** [comment]
I was thinking that an inline popup would be best - the whole idea behind it is that you aren't really going to keep it around for a long time, so not having it open a whole new browser window would be better. Not to mention that you may have multiple tabs with multiple sessions, and if you have real pop-ups separate from that it may be hard to keep track of which popup belongs to which session.

**Comment by chainsol** [comment]
I guess...every option? :D @j-f1, I've looked at that before, and I think we'll make it a little more dynamic for selecting popups. Some sort of registration function...

**Comment by friarzen** [comment]
Ok, after a quick EvMenu-code-analysis assist from Cloud_Keeper, I believe this can be addressed fairly trivially now.  The EvMenu self.caller.msg() calls would just need to set the type= kwarg to something predefined in the webclient (like, say, 'popup'?) and then have the popup plugin catch it (identical to how @help option-popup works in PR #1752) and open the content in a popup.

**Comment by Griatch** [comment]
I guess all that would be needed for this is offer a `popup` type that the gui understands. I don't think this is something that should be used by default though - popups are pretty annoying at the best of times ...

**Comment by friarzen** [comment]
Yeah, this doesn't need to be the default, popups could be a user option in the settings to turn on.
